packages/front-end/pages/settings/customfields.tsx

```diff
@@ -0,0 +1,258 @@
+            <div className="row mb-3 align-items-center">
+              <div className="col-auto">
+                <h3>Custom Experiment Fields</h3>
+                <p className="text-gray"></p>
+              </div>
+            </div>
+            <div className="row mb-3">
+              <div className="col-12 mb-2 py-2">
+                <p>
+                  Custom experiment fields are available with an Enterprise
+                  plan.
+                </p>
+                <UpgradeMessage
+                  showUpgradeModal={() => setUpgradeModal(true)}
+                  commercialFeature="override-metrics"
+                  upgradeMessage="set custom experiment fields"
+                  isEnterprise={true}
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+        {upgradeModal && (
+          <>
+            <UpgradeModal
+              close={() => setUpgradeModal(false)}
+              reason="To add custom experiment metadata,"
+              source="custom-fields"
+            />
+          </>
+        )}
+      </>
+    );
+  }
+
+  return (
+    <>
+      <div className="contents container-fluid pagecontents">
+        <div className="mb-5">
+          <div className="row mb-3 align-items-center">
+            <div className="col-auto">
+              <h3>Custom Experiment Fields</h3>
+              <p className="text-gray"></p>
+            </div>
+            <div style={{ flex: 1 }} />
+            {permissions.manageCustomFields && (
+              <div className="col-auto">
+                <button
+                  className="btn btn-primary float-right"
+                  onClick={() => {
+                    setModalOpen({});
+                  }}
+                >
+                  <span className="h4 pr-2 m-0 d-inline-block align-top">
+                    <GBEdit />
+                  </span>
+                  Add Custom Field
+                </button>
+              </div>
+            )}
+          </div>
+          <DndContext
+            sensors={sensors}
+            onDragEnd={handleDragEnd}
+            onDragStart={handleDragStart}
+            onDragCancel={handleDragCancel}
+            collisionDetection={closestCenter}
+          >
+            <table className="table gbtable appbox">
+              <thead>
+                <tr>
+                  <th style={{ width: "30px" }}></th>
+                  <th>Field Name</th>
+                  <th>Field Description</th>
+                  <th>Field Type</th>
+                  <th>Projects</th>
+                  <th>Required</th>
+                  <th style={{ width: 75 }}></th>
+                </tr>
+              </thead>
+              <tbody>
+                {customFields?.length ? (
+                  <SortableContext
+                    items={items}
+                    strategy={verticalListSortingStrategy}
+                  >
+                    {items.map((v) => {
```

Comment:
When creating a new custom field, I immediately get this error upon saving:
```
TypeError: Cannot read properties of undefined (reading 'map')
```
Guessing there's a slight delay for when items get set after customFields are set. Could add an additional check to your conditional render rule on line 194.